File: 240308/100 도달하기/reach-100.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
     cin>>n;
    int cnt;
    int arr[100]={1,n};
   

    while(true){
        cnt++;
        arr[cnt]=arr[cnt-1]+arr[cnt-2];
        if(arr[cnt]>100)
        break;
    }
    for(int i=0;i<=cnt;i++){
        cout<<arr[i]<<" ";
    }
    
    return 0;
}

File: 240528/100 도달하기/reach-100.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
    Scanner sc=new Scanner(System.in);

    int n=sc.nextInt();
    int[] arr=new int[100];
    arr[1]=1;
    arr[2]=n;
    int cnt=0;

    for(int i=3;i<100;i++){
        arr[i]=arr[i-1]+arr[i-2];
    }

    for(int i=1;i<100;i++){
        System.out.print(arr[i]+" ");
        if(arr[i]>100)
        break;
    }
    
    }
}